analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -123,6 +125,27 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
+    public bool ShouldAnalyzeFile(string filePath) =>
+        ProjectConfiguration().ProjectType != ProjectType.Unknown
+        || (FileInclusionCache.GetValue(Compilation, _ => new()) is var cache // SonarLint, NuGet or Scanner <= 5.0
```

Comment:
I find the comment a bit misleading on this line. It should most probably go on the line above, and in a negated context. as If the ProjectType is not Unknown it means it is not SL, Nuget or Scanner <=5.0